content/docs/reference-react-dom.md

```diff
@@ -55,6 +55,21 @@ createPortal(child, container)
 ```
 
 Creates a portal. Portals provide a way to [render children into a DOM node that exists outside the hierarchy of the DOM component](/docs/portals.html).
+
+### `flushSync()` {#flushsync}
+
+```javascript
+flushSync(callback)
+```
+
+Force React to flush updates synchronously. This method is useful for being able to read the value of `state` immediately.
```

Comment:
This a bit misleading since the easiest way to read the value of state immediately is just read what you set it to. I don't think we should encourage this as a workaround for that.

It's more about reading the outcomes. Almost always from the DOM.

It could use an explanation what the purpose of the callback is since callbacks are typically associated with async operation completing, which this by definition is not.

It's meant to form a grouping of any setStates that you need to flush synchronously.

Reply:
Makes sense, I was being a little value about the callback because it's not necessarily only the updates passed in that are flushed, correct? I.e. just calling `flushSync()` without a callback will flush work?

Reply:
Yea, if you happen to know that they're also scheduled in the sync lane. You can force any pending discrete ones to flush.

However, it's a little sketchy to assume that the context you're executing within is already discrete so adding more setStates before it might not be a good idea. Since you could've been wrapped in a startTransition.

```js
// Reasonable if that's your use case
flushSync();
```

```js
// Bad
setState(true);
flushSync();
```

```js
// Good
flushSync(() => setState(true));
```

Reply:
I'm trying to avoid explaining discrete updates since we haven't really explained it elsewhere, how about a note like:

> `flushSync` may also flush updates outside of the callback when necessary to flush the updates provided.

Reply:
Sounds good but that part might even need explanation.

The important part is that you put the updates you want to do inside the closure.